src/app/App.js

```diff
@@ -13,6 +19,21 @@ import {
 import "./App.css";
 
 function App() {
+  const { token } = useSelector((store) => store.authDetail);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (token) dispatch(getUserDetail(token));
+  }, [dispatch, token]);
+
+  useEffect(() => {
+    if (token) dispatch(getAllPosts());
+  }, [dispatch, token]);
+
+  useEffect(() => {
+    if (token) dispatch(getAllUsers());
+  }, [dispatch, token]);
+
```

Comment:
you can create custom hook for this

Reply:
Thank you, I will refractor it later on.